elasticdl/python/tests/task_manager_test.py

```diff
@@ -132,6 +134,36 @@ def test_invoke_train_end_callback(self):
             task_d._todo[0].type, elasticdl_pb2.TRAIN_END_CALLBACK
         )
 
+    def test_check_and_reassign_timeout_tasks(self):
+        task_manager = create_task_manager({"f1": (0, 10), "f2": (0, 10)}, {})
+        task_manager.create_tasks(elasticdl_pb2.TRAINING)
+        task_count = len(task_manager._todo)
+        task_start_time = time.time() - 1000
+        task_manager._worker_start_task_time[0] = task_start_time
+        task_manager._doing[0] = (0, task_manager._todo[0], task_start_time)
+        task_manager._todo.pop()
+
```

Comment:
We can use the following way in unit test to make sure the test always passes. Our goal is to check the the logic correctness of `_check_and_reassign_timeout_tasks`.
If we create a thread, there may be a rare case that the background thread is not scheduled.
```
time.sleep(1)
task_manager._check_and_reassign_timeout_tasks()
```

Reply:
But, there is an infinite loop in the `task_manager._check_and_reassign_timeout_tasks()`.  If we call `task_manager._check_and_reassign_timeout_tasks()`in the unittest and the unittest will not exit.